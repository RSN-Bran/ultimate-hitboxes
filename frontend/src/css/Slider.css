

@media (max-width: 1200px) {
  #videoSlider {
    width: 80%
  }
}

@media (min-width: 1200px) {
  #videoSlider {
    width: 100%
  }
}

#videoSlider::-webkit-slider-thumb {
  cursor: pointer;
}
