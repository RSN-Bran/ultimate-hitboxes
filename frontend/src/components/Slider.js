import * as React from "react"

import '../css/Slider.css';

function Slider(props) {

	if (props.totalFrames == 1) {
		return null;
	}

	//CSS, move to CSS file later
	let sliderStyle = {width: "100%"}

	//Create TickMarks for Range slider
	let tickMarks = []
	for (let i = 1; i <= props.totalFrames; i++) {
		tickMarks.push(<option key={i}>{i}</option>)
	}
	
	return (
		<div id="sliderContainer" style={sliderStyle}>
			<h5 id="frameIndicator">Frame: {props.currentFrame}/{props.totalFrames}</h5>
			<div>
				<input
					id="videoSlider"
					name="videoSlider"
					type="range"
					min="1"
					max={props.totalFrames}
					value={props.currentFrame}
					onChange={e => props.jumpToFrame(parseInt(e.target.value))}
					list="ticks"
				/>
			</div>
			<datalist id="ticks">
				{tickMarks}
			</datalist>
		</div>
	)
	
}

export default Slider