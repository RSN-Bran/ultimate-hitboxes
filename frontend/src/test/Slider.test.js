import React from 'react'
//import {render} from "@testing-library/react"
import {create, act} from 'react-test-renderer';
import {shallow} from 'enzyme'

import Slider from '../components/Slider'


import "@testing-library/jest-dom/extend-expect"
import { Router } from 'react-router-dom'

describe ('Slider component', () => {
    it('current frame test', () => {
        let currentFrame=3
        let totalFrames=10
        const wrapper = shallow(<Slider currentFrame={currentFrame} totalFrames={totalFrames}/>)
        const text = wrapper.find('h5').text()
        expect(text).toEqual(`Frame: ${currentFrame}/${totalFrames}`)
    })

    it('total frame is 1', () => {
        let currentFrame=1
        let totalFrames=1
        const wrapper = shallow(<Slider currentFrame={currentFrame} totalFrames={totalFrames}/>)
        expect(wrapper.isEmptyRender()).toBe(true)
    })

    it('user interaction', () => {
        let func = (frame) => {
            currentFrame = frame
        }
        let initialFrame=1
        let currentFrame=1
        let totalFrames=10
        const wrapper = shallow(<Slider currentFrame={currentFrame} totalFrames={totalFrames} jumpToFrame={func}/>)
        
        wrapper.find('#videoSlider').simulate('change', {target: {value:currentFrame+1}})

        expect(currentFrame).toEqual(initialFrame+1)
    })
})

test('Slider Snapshot Test', () => {

    let func = (frame) => {
        console.log(frame)
    }

    let app;
    act(()=>{
        app = create(<Slider totalFrames={11} currentFrame={2} jumpToFrame={func}/>)
    })


    expect(app.toJSON()).toMatchSnapshot()
    
})