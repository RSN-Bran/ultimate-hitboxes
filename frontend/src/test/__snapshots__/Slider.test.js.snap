// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Snapshot Test 1`] = `
<div
  id="sliderContainer"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <h5
    id="frameIndicator"
  >
    Frame: 
    2
    /
    11
  </h5>
  <div>
    <input
      id="videoSlider"
      list="ticks"
      max={11}
      min="1"
      name="videoSlider"
      onChange={[Function]}
      type="range"
      value={2}
    />
  </div>
  <datalist
    id="ticks"
  >
    <option>
      1
    </option>
    <option>
      2
    </option>
    <option>
      3
    </option>
    <option>
      4
    </option>
    <option>
      5
    </option>
    <option>
      6
    </option>
    <option>
      7
    </option>
    <option>
      8
    </option>
    <option>
      9
    </option>
    <option>
      10
    </option>
    <option>
      11
    </option>
  </datalist>
</div>
`;
